### DISCUSSION TITLE
Is Idyll still maintained?

### DISCUSSION THREAD

**Discussion by jimidle** [idyll-lang/idyll#787] [General]
The package seems to have fallen well behind in dependencies, and trying to create new React components is quite difficult when .js components are the only thing available (no .tsx support). I am having a real hard time trying to get any React charting to work as I just get hit with the usual npm dependency hell.

So, I wonder if it has been left to fallow?

**Comment by mathisonian** [comment]
Hi @jimidle, Idyll is still maintained albeit things have been quite slow the last few months due to other commitments. 

I've been working on the 6.0 release which you can currently download via `npm i idyll@6.0.0-alpha.6`. This includes a bunch of internal improvements, including switching our build system to use esbuild which gives nice performance and compatibility improvements. 

Typescript support isn't currently on the roadmap but I'm not opposed to adding it. Would you mind opening an issue for that and we can discuss details there?

**Reply by jimidle** [reply]
Thank you for taking the time to reply. It is good to hear you are working on it still, as it is a fantastic system for technical presentations. I appreciate that we all have many other things to do than work on software we are giving away for free. 

I tried the v6 beta, but that is where I started running into dependency conflicts etc. When I went back to the latest stable release, everything worked so long as I didn't try to upgrade anything ;)

I will raise some enhancement requests and a few _issues_, which may in fact me being a backend programmer and not knowing enough about browsers and React etc. Feel free to correct me or close any issues. I will try not to raise trivial things that I can find the answers to elsewhere.

**Reply by mathisonian** [reply]
Thanks @jimidle! yes - please don't be shy about opening issues as you run into things. I've commented on the ones you've raised so far. 

I'd like to figure out what the dependency issues you are hitting in the v6 beta are - can you share which libraries you were trying to load or any other steps to reproduce?

**Reply by jimidle** [reply]
I will start afresh with an Alpha 6 project and write down what I do. It could be because I was trying to use external components such as echarts-for-react and maybe got lost along the way. Let me eliminate my own incomplete knowledge of npm etc first i guess ;)
